### PR TITLE
Arms warrior

### PR DESCRIPTION
**Pull request by BadBossy**
Added the basic Layout for Arms Warrior, more is coming, but this is just a simple version. Need some suggestions about what to add further soon.

What did I do?

- Initial Setup
- Colossus Smash Uptime
- DPS
- Downtime
- Spells added

Issue: #570

My Discord: Bossy#9968

### CODE REVIEW

File: src/Parser/Warrior/Arms/CHANGELOG.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+export default `
+16-10-2017 - Added a Light of Dawn average players hit statistic. (by Zerotorescue)
```
Comment (anom0ly) [inline src/Parser/Warrior/Arms/CHANGELOG.js]: This should be cleaned up and only contain references to Arms warrior module

File: src/CHANGELOG.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export default `
+19-10-2017 - Added Arms Warrior Initial Setup (by TheBadBossy)
```
Comment (anom0ly) [inline src/CHANGELOG.js]: No real need to add this. Just use the spec specific `CHANGELOG.js` file

File: src/Parser/Warrior/Arms/CONFIG.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  maintainer: '@TheBadBossy',
+  maintainerAvatar: thebadbossy_avatar,
+  description: (
+    <div>
+      Hey I've been hard at work making this analyzer for you. I hope the suggestions give you useful pointers to improve your performance. Remember: focus on improving only one or two important things at a time. Improving isn't easy and will need your full focus until it becomes second nature to you.<br /><br />
+      </div>
+  ),
+  // When changing this please make a PR with ONLY this value changed, we will do a review of your analysis to find out of it is complete enough.
+  completeness: SPEC_ANALYSIS_COMPLETENESS.GREAT,
+  specDiscussionUrl: 'https://github.com/WoWAnalyzer/WoWAnalyzer/issues/54',
```
Comment (anom0ly) [inline src/Parser/Warrior/Arms/CONFIG.js]: Both line 22 and 23 just need to be updated with the spec specific information.

`SPEC_ANALYSIS_COMPLETENESS.NEEDS_MORE_WORK` should be used for now.

File: src/Parser/Warrior/Arms/Modules/BuffDebuff/ColossusSmashUptime.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  static dependencies = {
+    enemies: Enemies,
+  };
+
+  suggestions(when) {
+    const colossusSmashUptime = this.enemies.getBuffUptime(SPELLS.COLOSSUS_SMASH_DEBUFF.id) / this.owner.fightDuration;
+
+    when(colossusSmashUptime).isLessThan(0.95)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<span>Your <SpellLink id={SPELLS.COLOSSUS_SMASH_DEBUFF.id} /> uptime can be improved. Try to pay more attention to your debuff on the Boss</span>)
```
Comment (anom0ly) [inline src/Parser/Warrior/Arms/Modules/BuffDebuff/ColossusSmashUptime.js]: Can you expand on just why it is important to keep the debuff up? Does is provide increased damage / used for additional procs / etc?

File: src/Parser/Warrior/Arms/CombatLogParser.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+import CoreCombatLogParser from 'Parser/Core/CombatLogParser';
+import DamageDone from 'Parser/Core/Modules/DamageDone';
+
+import CastEfficiency from './Modules/Features/CastEfficiency';
+import AlwaysBeCasting from './Modules/Features/AlwaysBeCasting';
+import CooldownTracker from './Modules/Features/CooldownTracker';
+import ColossusSmashUptime from './Modules/BuffDebuff/ColossusSmashUptime';
+
+//import RelicTraits from './Modules/Traits/RelicTraits';
```
Comment (anom0ly) [inline src/Parser/Warrior/Arms/CombatLogParser.js]: Any reason Relic traits are disabled?

File: src/Parser/Warrior/Arms/Modules/Features/CastEfficiency.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+class CastEfficiency extends CoreCastEfficiency {
+  static CPM_ABILITIES = [
+    ...CoreCastEfficiency.CPM_ABILITIES,
+    {
+      spell: SPELLS.BATTLE_CRY,
+      category: CastEfficiency.SPELL_CATEGORIES.COOLDOWNS,
+      getCooldown: haste => 60,
+      noSuggestion: true,
+    },
```
Comment (anom0ly) [inline src/Parser/Warrior/Arms/Modules/Features/CastEfficiency.js]: Potentially want to work to expand this area within the first PR just to give some additional context for the user on what they are casting, or casts they are missing.

File: src/Parser/Warrior/Arms/CombatLogParser.js (hunk 5 above)
Comment (BadBossy) [reply src/Parser/Warrior/Arms/CombatLogParser.js]: Due to long research and analyzing about which is usefull to show, I'm going to work on it in the next PR if you allow it.

Comment (anom0ly) [reply src/Parser/Warrior/Arms/CombatLogParser.js]: That is fine. Just was asking is all.

**Comment by BadBossy** [comment]
@anom0ly - updates and your suggestions done -> another review please and thanks! more is on the way -> relict traits, T20, Bladestorm/Ravenger, Proc usage

File: src/Parser/Warrior/Arms/CONFIG.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  maintainer: '@TheBadBossy',
+  maintainerAvatar: thebadbossy_avatar,
+  description: (
+    <div>
+      Hey I've been hard at work making this analyzer for you. I hope the suggestions give you useful pointers to improve your performance. Remember: focus on improving only one or two important things at a time. Improving isn't easy and will need your full focus until it becomes second nature to you.<br /><br />
+      </div>
+  ),
+  // When changing this please make a PR with ONLY this value changed, we will do a review of your analysis to find out of it is complete enough.
+  completeness: SPEC_ANALYSIS_COMPLETENESS.NEEDS_MORE_WORK,
+  specDiscussionUrl: 'https://github.com/WoWAnalyzer/WoWAnalyzer/issues/571',
```
Comment (sMteX) [inline src/Parser/Warrior/Arms/CONFIG.js]: After your PR (in the next perhaps, looks like it's gonna be soon) we'll create a spec discussion issue which link should be here instead.

File: src/Parser/Warrior/Arms/Modules/Features/CastEfficiency.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      getCooldown: haste => 60,
+      noSuggestion: false,
+    },
+    {
+      spell: SPELLS.CHARGE,
+      category: CastEfficiency.SPELL_CATEGORIES.UTILITY,
+      getCooldown: haste => 20,
+      recommendedCastEfficiency: 0.1,
+      noSuggestion: false,
+    },
```
Comment (sMteX) [inline src/Parser/Warrior/Arms/Modules/Features/CastEfficiency.js]: I think you'll have to add all other abilities (including rarely used utility spells) here in the very next PR, but it's enough for a starter.

**Pull request merged**